pkg/workloadapi/workload.proto

```diff
@@ -0,0 +1,101 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+option go_package="pkg/workloadapi";
+
+message Workload {
+  // Name represents the name for the workload. This may be elided as an optimization
+  string name = 1;
+  // Namespace represents the namespace for the workload. This may be elided as an optimization
+  string namespace = 8;
+
+  string canonical_name = 14;
+  string canonical_revision = 15;
+  string workload_type = 16;
+  string workload_name = 17;
+
+  // Address represents the IPv4/IPv6 address for the workload.
+  // This should be globally unique.
+  // This should not have a port number.
+  // TODO: Add network as discriminator
+  string address = 2;
+
+  // The SPIFFE identity of the workload.
+  string identity = 3;
+
+  // The protocol that should be used to connect to this workload.
+  Protocol protocol = 4;
+
+  // The address this workload can be reached at.
+  // If not set, address should be reached directly
+  string gateway_address = 5;
+  string remote_proxy = 11;
+  // Temporary solution:
+  // Set to disable node -> node direct calling
+  bool native_hbone = 13;
+
+  // Virtual IPs defines a set of virtual IP addresses the workload can be reached at,
+  map<string, PortList> virtual_ips = 6;
```

Comment:
cc @costinm This looks like a dependency on services as well, not just endpoints.

Reply:
That's fine - it's info about the endpoints, we're not trying to normalize a database. 

What I would like in the comments is what the field is used for, and if we can improve the representation to reduce 
object allocs/size/etc - since the goal is to have an optimized case-specific proto, not to reuse abstract/generic protos. 

For example do we need the port list ? Do we need the fully qualified service or just the base name ( services can't select cross ns ) ? Do we want the service VIP as well ( or what does virtual ips mean ? )

Reply:
We only have the IP, no names of anything. I update the docs to be more clear.

Port list - we might be able to get rid of it in some cases. We do need to know if there is service_port!=target_port though.

Reply:
I think we need a lot more documentation - for each field, where is it used and what it can be.

We only have IP - but do we send mixed Services VIPs and Pod IPs (and external IPs from ServiceEntry ) ? Any case where we'll let k8s do the LB and the VIP will be used ? I'm not entirely sure what you capture in ztunnel, I assume it's primarily service VIPs ?  

Since we're dumping all IPs - it is fine to mix them I think. Pod IPs have the list of service names, services IPs have the service as well - ztunnel can join them.